src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -109,34 +114,41 @@ uct_cuda_copy_post_cuda_async_copy(uct_ep_h tl_ep, void *dst, void *src,
     ucs_memory_type_t src_type;
     ucs_memory_type_t dst_type;
     CUstream *stream;
+    CUcontext current_ctx;
     ucs_queue_head_t *event_q;
+    uct_cuda_copy_per_ctx_rsc_t *ctx_rsc;
 
     if (!length) {
         return UCS_OK;
     }
 
+
     /* ensure context is set before creating events/streams */
-    if (iface->cuda_context == NULL) {
-        UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&iface->cuda_context));
-        if (iface->cuda_context == NULL) {
-            ucs_error("attempt to perform cuda memcpy without active context");
+    UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&current_ctx));
+    if (current_ctx == NULL) {
+        ucs_error("attempt to perform cuda memcpy without active context");
+        return UCS_ERR_IO_ERROR;
```

Comment:
Should we return error or attempt to set the context as we have the buffers? Though, it may not be in the scope of this PR.

Reply:
Scope for this PR is for the thread to have set the right context. Will address this in a follow up PR.